src/top/utop.el

```diff
@@ -207,13 +207,30 @@ before the end of prompt.")
   "The position of the cursor in the phrase sent to OCaml (where
 to add the newline character if it is not accepted).")
 
-(defvar utop-package-list nil
-  "List of packages to load when visiting OCaml buffer.
-Useful as file variable.")
+(make-variable-buffer-local
+ (defvar utop-package-list nil
+   "List of packages to load when visiting OCaml buffer.
+Useful as file variable."))
 
-(defvar utop-ocaml-preprocessor nil
-  "Name of preprocesor. Currently supported camlp4o, camlp4r.
-Useful as file variable.")
+(make-variable-buffer-local
+ (defvar utop-ocaml-preprocessor nil
+   "Name of preprocesor. Currently supported camlp4o, camlp4r.
+Useful as file variable."))
+
+(defvar utop-skip-blank-and-comments 'tuareg-skip-blank-and-comments
```

Comment:
It'd be nice if the default value was a function that dispatch according to the current mode to handle tuareg, caml and typerex. It doesn't seem too bad and make the transition easier for people who use either caml or typerex. Same for the next two variables.